src/components/interpolatedColorLegend.ts

```diff
@@ -0,0 +1,249 @@
+      this._upperLabel = this._content.append("g").classed(InterpolatedColorLegend.LEGEND_LABEL_CLASS, true);
+
+      this._measurer = new SVGTypewriter.Measurers.Measurer(this._content);
+      this._wrapper = new SVGTypewriter.Wrappers.Wrapper();
+      this._writer = new SVGTypewriter.Writers.Writer(this._measurer, this._wrapper);
+    }
+
+    public _requestedSpace(offeredWidth: number, offeredHeight: number): _SpaceRequest {
+      var textHeight = this._measurer.measure().height;
+
+      var ticks = this._scale.ticks();
+      var numSwatches = ticks.length;
+
+      var domain = this._scale.domain();
+      var labelWidths = domain.map((d: number) => this._measurer.measure(this._formatter(d)).width);
+
+      var desiredHeight: number;
+      var desiredWidth: number;
+      if (this._isVertical()) {
+        var longestWidth = _Util.Methods.max(labelWidths, 0);
+        desiredWidth = this._padding + textHeight + this._padding + longestWidth + this._padding;
+        desiredHeight = this._padding + numSwatches * textHeight + this._padding;
+      } else {
+        var totalWidth = labelWidths[0] + labelWidths[1];
+        desiredHeight = this._padding + textHeight + this._padding;
+        desiredWidth = this._padding + labelWidths[0] + this._padding
+                        + numSwatches * textHeight
+                        + this._padding + labelWidths[1] + this._padding;
+      }
+
+      return {
+        width : desiredWidth,
+        height: desiredHeight,
+        wantsWidth: offeredWidth < desiredWidth,
+        wantsHeight: offeredHeight < desiredHeight
+      };
+    }
+
+    private _isVertical() {
+      return this._orientation !== "horizontal";
+    }
+
+    public _doRender() {
+      super._doRender();
+
+      var domain = this._scale.domain();
+
+      var textHeight = this._measurer.measure().height;
+      var text0 = this._formatter(domain[0]);
+      var text0Width = this._measurer.measure(text0).width;
+      var text1 = this._formatter(domain[1]);
+      var text1Width = this._measurer.measure(text1).width;
+
+      var ticks = this._scale.ticks();
+      var numSwatches = ticks.length;
+
+      var padding = this._padding;
+
+      var upperLabelShift: Point = { x: 0, y: 0 };
+      var lowerLabelShift: Point = { x: 0, y: 0 };
+      var lowerWriteOptions = {
+                selection: this._lowerLabel,
+                xAlign: "center",
+                yAlign: "center",
+                textRotation: 0
+            };
+      var upperWriteOptions = {
+                selection: this._upperLabel,
+                xAlign: "center",
+                yAlign: "center",
+                textRotation: 0
+            };
+
+      var swatchWidth: number;
+      var swatchHeight: number;
+      var swatchX: (d: any, i: number) => number;
+      var swatchY: (d: any, i: number) => number;
+
+      if (this._isVertical()) {
+        var longestTextWidth = text1Width > text0Width ? text1Width : text0Width;
```

Comment:
I thought about that, but the behavior is poorly defined if either end of the scale is `undefined`. Is that okay?